### ISSUE TITLE
Discovered instance - login didn't work

### ISSUE THREAD

**Issue by andriej** [home-assistant/android#312]
**Home Assistant Android version:** 1.5.0 Google Play

**Android version:** 10

**Phone model:** OnePlus6

**Home Assistant version:** 0.104.3

**Last working Home Assistant release (if known):** n/a

**Description of problem:** Discovered instance of HA while trying to login (I saw name of it and proper URL address with https://) made me thru login page and 2FA and then I saw a message about error ("couldn't load"). There were although no issues when I entered the URL by hand.
I use own custom domain with LE SSL and locally it's translated by router directly to HA.
So in short: proper credentials and proper 2FA code made it failed to login/register.

**Traceback (if applicable):**
n/a

**Additional information:**
